I tried that. npm install fails for me in both cases

```
❯ npm install
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: coc.nvim@0.0.83-next.18
npm error Found: @types/node@14.18.63
npm error node_modules/@types/node
npm error   dev @types/node@"^14.14" from the root project
npm error   @types/node@"*" from @types/fs-extra@8.1.5
npm error   node_modules/@types/fs-extra
npm error     dev @types/fs-extra@"^8.0.0" from the root project
npm error   1 more (@types/glob)
npm error
npm error Could not resolve dependency:
npm error peer @types/node@"^16.18.0" from coc.nvim@0.0.83-next.18
npm error node_modules/coc.nvim
npm error   dev coc.nvim@"^0.0.83-next.17" from the root project
npm error
npm error Conflicting peer dependency: @types/node@16.18.97
npm error node_modules/@types/node
npm error   peer @types/node@"^16.18.0" from coc.nvim@0.0.83-next.18
npm error   node_modules/coc.nvim
npm error     dev coc.nvim@"^0.0.83-next.17" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /home/jan/.npm/_logs/2024-05-25T08_03_30_253Z-eresolve-report.txt
npm error A complete log of this run can be found in: /home/jan/.npm/_logs/2024-05-25T08_03_30_253Z-debug-0.log
```

What I am looking for is a simple notification of the missing ``compile_commands.json`` file in the status bar without any prompt. I also don't think that the option to autogenerate the ``coc-settings.json`` makes much sense. For me it genertated that in the project root directory and suddenly I had a hidden folder there that was added to my version control by mistake. The user should configure the plugin manually as provided by the readme.

--- a/README.md
+++ b/README.md
@@ -62,6 +62,14 @@ This extension also provides a custom root directory where the sonar lint binari
 }
 ```
 
+Sonarlint will try to prompt you when a compile commands in c or cpp projects is not found, to disable that feature globally, and ignore the missing compile commands you can set the following property in your coc-settings.json
+
+```json
+{
+    "sonarlint.notifyMissingCompileCommands": false
+}
+```
+
 The directory must contain two folders, a server/ and analyzers/, where the server binary is located in the server/ folder and all analyzer binaries are located in the analyzers/ folder
 
 ### JS/TS analysis specific requirements

--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
     "name": "coc-sonarlint",
     "version": "0.0.10",
     "description": "sonarlint for coc.nvim",
-    "author": "asmodeus812 <svetlozar321@gmail.com>",
+    "author": "asmodeus812",
     "license": "MIT",
     "main": "lib/index.js",
     "keywords": [
@@ -308,6 +308,13 @@
                     "type": "string",
                     "default": "",
                     "markdownDescription": "Files whose name match this [glob pattern](https://docs.oracle.com/javase/tutorial/essential/io/fileOps.html#glob) will not be processed by analyzers. In [Connected Mode](command:SonarLint.HelpAndFeedbackLinkClicked?%22connectedModeDocs%22) with SonarQube or SonarCloud, this property will be ignored and server exclusion settings will be applied. No rules are evaluated on excluded files. Example: `**/lib/**,**/*generated*`",
+                    "scope": "resource"
+                },
+                "sonarlint.notifyMissingCompileCommands": {
+                    "order": 21,
+                    "default": true,
+                    "type": "boolean",
+                    "markdownDescription": "Specifies if missing or invalid compile commands in c or cpp projects will prompt the user to generate or create one, by default the user will be asked for each new workspace he visits, and the user can opt out of it per workspace by selecting the option - Do not ask for compile commands again for this workspace",
                     "scope": "resource"
                 },
                 "sonarlint.ls.javaHome": {

--- a/src/cfamily/cfamily.ts
+++ b/src/cfamily/cfamily.ts
@@ -9,12 +9,9 @@
 import * as coc from "coc.nvim"
 import * as fs from "fs"
 import * as path from "path"
-import {SONARLINT_CATEGORY} from "../settings/settings"
+import {isNotificationEnabled, updateCompileCommandsPath, updateNotificationDisabled} from "../settings/settings"
 import {SonarLintDocumentation} from "../commons"
 
-const PATH_TO_COMPILE_COMMANDS = "pathToCompileCommands"
-const FULL_PATH_TO_COMPILE_COMMANDS = `${SONARLINT_CATEGORY}.${PATH_TO_COMPILE_COMMANDS}`
-const DO_NOT_ASK_ABOUT_COMPILE_COMMANDS_FLAG = "doNotAskAboutCompileCommands"
 let remindMeLaterAboutCompileCommandsFlag = false
 let userSelectingCompileCommandsChoice = false
 
@@ -22,27 +19,8 @@ function showMessageAndUpdateConfig(compilationDbPath: string) {
     coc.window.showInformationMessage(
         `Analysis configured. Compilation database path is set to: ${compilationDbPath}`,
     )
-    const [pathForSettings, workspaceFolder] =
-        tryRelativizeToWorkspaceFolder(compilationDbPath)
-
-    if (workspaceFolder !== undefined) {
-        const config = coc.workspace.getConfiguration(
-            SONARLINT_CATEGORY,
-            workspaceFolder.uri,
-        )
-        return config.update(
-            PATH_TO_COMPILE_COMMANDS,
-            pathForSettings,
-            coc.ConfigurationTarget.WorkspaceFolder,
-        )
-    }
-    return coc.workspace
-        .getConfiguration()
-        .update(
-            FULL_PATH_TO_COMPILE_COMMANDS,
-            pathForSettings,
-            coc.ConfigurationTarget.Workspace,
-        )
+    const [pathForSettings, _] = tryRelativizeToWorkspaceFolder(compilationDbPath)
+    updateCompileCommandsPath(pathForSettings)
 }
 
 function tryRelativizeToWorkspaceFolder(filePath: string): [string, coc.WorkspaceFolder | undefined] {
@@ -59,68 +37,51 @@ function tryRelativizeToWorkspaceFolder(filePath: string): [string, coc.Workspac
     return [filePath, undefined]
 }
 
-export async function configureDoNotAskCompilationDatabase(context: coc.ExtensionContext) {
-    return context.workspaceState.update(
-        DO_NOT_ASK_ABOUT_COMPILE_COMMANDS_FLAG,
-        true,
-    )
-}
-
-export async function configureCompilationDatabase(context: coc.ExtensionContext) {
+export async function configureCompilationDatabase() {
     const paths = (await coc.workspace.findFiles(`**/compile_commands.json`)).filter((path) => fs.existsSync(path.fsPath))
     if (paths.length === 0) {
         coc.window.showWarningMessage(`No compilation databases were found in the workspace\n [How to generate compile commands](${SonarLintDocumentation.C_CPP_ANALYSIS})`)
-        await coc.workspace
-            .getConfiguration()
-            .update(
-                FULL_PATH_TO_COMPILE_COMMANDS,
-                undefined,
-                coc.ConfigurationTarget.Workspace,
-            )
+        updateCompileCommandsPath(undefined)
     } else {
         await showCompilationDatabaseOptions(paths)
     }
 }
 
-export function notifyMissingCompileCommands(context: coc.ExtensionContext) {
+export function notifyMissingCompileCommands() {
     return async () => {
-        if (userSelectingCompileCommandsChoice || (await doNotAskAboutCompileCommandsFlag(context)) || remindMeLaterAboutCompileCommandsFlag) {
+        if (userSelectingCompileCommandsChoice || (isNotificationEnabled() !== true) || remindMeLaterAboutCompileCommandsFlag) {
             return
         }
         userSelectingCompileCommandsChoice = true
+        const remindMeLaterAction = "Ask me later again"
+        const doNotAskGlobally = "Never ask or prompt me again"
+        const doNotAskForThisWorkspace = "Do not ask for this workspace"
         const configureCompileCommandsAction = "Configure compile commands now"
-        const remindMeLaterAction = "Ask me later to configure compile commands"
-        const doNotAskAboutCompileCommandsAction = "Do not ask for compile commands again for this workspace"
         const message = `SonarLint might be unable to analyze C and C++ file(s) because there is no configured compilation database.`
         const selection = await coc.window
             .showWarningMessage(
                 message,
                 configureCompileCommandsAction,
                 remindMeLaterAction,
-                doNotAskAboutCompileCommandsAction
+                doNotAskForThisWorkspace,
+                doNotAskGlobally
             )
         switch (selection) {
             case configureCompileCommandsAction:
-                configureCompilationDatabase(context)
+                configureCompilationDatabase()
                 break
             case remindMeLaterAction:
                 remindMeLaterAboutCompileCommandsFlag = true
                 break
-            case doNotAskAboutCompileCommandsAction:
-                configureDoNotAskCompilationDatabase(context)
+            case doNotAskForThisWorkspace:
+                updateNotificationDisabled(false, undefined);
+                break
+            case doNotAskGlobally:
+                updateNotificationDisabled(false, coc.ConfigurationTarget.Global);
                 break
         }
         userSelectingCompileCommandsChoice = false
     }
-}
-
-async function doNotAskAboutCompileCommandsFlag(
-    context: coc.ExtensionContext,
-): Promise<boolean> {
-    return context.workspaceState.get(
-        DO_NOT_ASK_ABOUT_COMPILE_COMMANDS_FLAG,
-        false,
-    )
 }
 
 async function showCompilationDatabaseOptions(paths: coc.Uri[]) {

--- a/src/index.ts
+++ b/src/index.ts
@@ -7,13 +7,13 @@
 import * as util from "./util/util"
 import * as protocol from "./lsp/protocol"
 import * as ChildProcess from "child_process"
-import type { Position, Location } from "vscode-languageserver-types"
-import { ExtensionContext, StreamInfo, window } from "coc.nvim"
+import type {Position, Location} from "vscode-languageserver-types"
+import {ExtensionContext, StreamInfo, window} from "coc.nvim"
 import * as coc from "coc.nvim"
 import {
-    enableVerboseLogs,
+    updateVerboseLogging,
     isVerboseEnabled,
-    loadInitialSettings,
+    isNotificationEnabled
 } from "./settings/settings"
 import {
     getLogOutput,
@@ -21,12 +21,12 @@ import {
     logToSonarLintOutput,
     showLogOutput,
 } from "./util/logging"
-import { setExtensionContext } from "./util/util"
-import { languageServerCommand } from "./lsp/server"
-import { JAVA_HOME_CONFIG, installManagedJre, resolveRequirements } from "./util/requirements"
-import { SonarLintExtendedLanguageClient } from "./lsp/client"
-import { getPlatform } from "./util/platform"
-import { Commands } from "./util/commands"
+import {setExtensionContext} from "./util/util"
+import {languageServerCommand} from "./lsp/server"
+import {JAVA_HOME_CONFIG, installManagedJre, resolveRequirements} from "./util/requirements"
+import {SonarLintExtendedLanguageClient} from "./lsp/client"
+import {getPlatform} from "./util/platform"
+import {Commands} from "./util/commands"
 import {
     AllRulesTreeDataProvider,
     LanguageNode,
@@ -34,8 +34,8 @@ import {
     languageKeyDeNormalization,
     toggleRule,
 } from "./rules/rules"
-import { getJavaConfig, installClasspathListener } from "./java/java"
-import { showRuleDescription } from "./rules/rulepanel"
+import {getJavaConfig, installClasspathListener} from "./java/java"
+import {showRuleDescription} from "./rules/rulepanel"
 import {
     configureCompilationDatabase,
     notifyMissingCompileCommands,
@@ -43,7 +43,7 @@ import {
 } from "./cfamily/cfamily"
 
 const DOCUMENT_SELECTOR = [
-    { scheme: "file", pattern: "**/*" },
+    {scheme: "file", pattern: "**/*"},
     {
         notebook: {
             scheme: "file",
@@ -76,7 +76,7 @@ async function runJavaServer(
         })
         .then((requirements) => {
             return new Promise<StreamInfo>((resolve, reject) => {
-                const { command, args }: any = languageServerCommand(context, requirements)
+                const {command, args}: any = languageServerCommand(context, requirements)
                 if (!command) {
                     reject(new Error("Failed to resolve launch command and args"))
                     return
@@ -84,7 +84,7 @@ async function runJavaServer(
                 logToSonarLintOutput(`Executing ${command} ${args.join(" ")}`)
                 const process = ChildProcess.spawn(command, args)
 
-                process.stderr.on("data", function(data) {
+                process.stderr.on("data", function (data) {
                     logWithPrefix(data, "[stderr]")
                 })
 
@@ -97,8 +97,6 @@ async function runJavaServer(
 }
 
 export async function activate(context: ExtensionContext): Promise<void> {
-
-    loadInitialSettings()
     setExtensionContext(context)
     initLogOutput(context)
 
@@ -237,7 +235,7 @@ function registerCommands(context: coc.ExtensionContext) {
                 const indexOfSeparator = ruleKey.indexOf(":")
                 const language = indexOfSeparator > 0 ? ruleKey.substring(0, indexOfSeparator) : null
                 const type = indexOfSeparator > 0 && language
-                    ? new RuleNode({ key: languageKeyDeNormalization(language) + ":" + ruleKey.substring(indexOfSeparator + 1).toLowerCase() } as protocol.Rule, language)
+                    ? new RuleNode({key: languageKeyDeNormalization(language) + ":" + ruleKey.substring(indexOfSeparator + 1).toLowerCase()} as protocol.Rule, language)
                     : new LanguageNode(ruleKey, coc.TreeItemCollapsibleState.Collapsed)
                 let node = await allRulesTreeDataProvider.getTreeItem(type)
                 if (type instanceof RuleNode && language) {
@@ -247,11 +245,11 @@ function registerCommands(context: coc.ExtensionContext) {
                         await allRulesTreeDataProvider.getChildren(pnode)
                         node = await allRulesTreeDataProvider.getTreeItem(type)
                     }
-                    await allRulesView?.reveal(node, { select: true, focus: true, expand: true })
+                    await allRulesView?.reveal(node, {select: true, focus: true, expand: true})
                 } else if (type instanceof LanguageNode) {
                     node.collapsibleState = coc.TreeItemCollapsibleState.Expanded
                     allRulesTreeDataProvider.register(node)
-                    await allRulesView?.reveal(node, { select: true, focus: true, expand: true })
+                    await allRulesView?.reveal(node, {select: true, focus: true, expand: true})
                 } else {
                     coc.window.showWarningMessage(`Unable to find or resolve rule id ${ruleKey}`)
                 }
@@ -286,22 +284,20 @@ function registerCommands(context: coc.ExtensionContext) {
     context.subscriptions.push(
         coc.commands.registerCommand(
             Commands.CONFIGURE_COMPILATION_DATABASE,
-            configureCompilationDatabase,
+            () => void configureCompilationDatabase(),
         ),
     )
 
     context.subscriptions.push(
-        coc.commands.registerCommand(Commands.ENABLE_VERBOSE_LOGS, () =>
-            enableVerboseLogs(),
-        ),
+        coc.commands.registerCommand(Commands.ENABLE_VERBOSE_LOGS, () => updateVerboseLogging(true))
     )
 
     context.subscriptions.push(
         coc.commands.registerCommand(Commands.INSTALL_MANAGED_JRE, () => {
-            installManagedJre(context, function() {
+            installManagedJre(context, function () {
                 coc.window
                     .showInformationMessage(`Downloaded & installed a managed jre`)
-            }, function() {
+            }, function () {
                 coc.window
                     .showErrorMessage(`Unable to download managed jre`)
             })
@@ -328,7 +324,7 @@ function installCustomRequestHandlers(context: coc.ExtensionContext) {
         util.shouldAnalyseFile(params.uri),
     )
     languageClient.onRequest(
-        protocol.CanShowMissingRequirementNotification.type, () => { return true })
+        protocol.CanShowMissingRequirementNotification.type, () => {return true})
     languageClient.onNotification(
         protocol.ShowSonarLintOutputNotification.type,
         () => void coc.commands.executeCommand(Commands.SHOW_SONARLINT_OUTPUT),
@@ -351,10 +347,13 @@ function installCustomRequestHandlers(context: coc.ExtensionContext) {
             coc.Uri.parse(browseTo),
         ),
     )
-    languageClient.onNotification(
-        protocol.NeedCompilationDatabaseRequest.type,
-        notifyMissingCompileCommands(context),
-    )
+
+    if (isNotificationEnabled() !== false) {
+        languageClient.onNotification(
+            protocol.NeedCompilationDatabaseRequest.type,
+            notifyMissingCompileCommands(),
+        )
+    }
 }
 
 async function showAllLocations(issue: protocol.Issue) {
@@ -372,7 +371,7 @@ async function showAllLocations(issue: protocol.Issue) {
                     character: textRange.endLineOffset,
                 } as Position,
             }
-            locations.push({ uri: loc.uri, range: range } as Location)
+            locations.push({uri: loc.uri, range: range} as Location)
         }),
     )
 

--- a/src/settings/settings.ts
+++ b/src/settings/settings.ts
@@ -18,30 +18,31 @@ let currentConfig: WorkspaceConfiguration
 
 export const SONARLINT_CATEGORY = "sonarlint"
 export const VERBOSE_LOGS = "output.showVerboseLogs"
+export const PATH_TO_COMPILE_COMMANDS = "pathToCompileCommands"
+export const NOTIFY_COMPILE_COMMANDS = "notifyMissingCompileCommands"
 
 export function getSonarLintConfiguration(): WorkspaceConfiguration {
     return workspace.getConfiguration(SONARLINT_CATEGORY)
 }
 
 export function isVerboseEnabled(): boolean {
-    return getCurrentConfiguration()?.get(VERBOSE_LOGS, false)
+    return getSonarLintConfiguration().get(VERBOSE_LOGS, false)
 }
 
-export function enableVerboseLogs() {
-    getCurrentConfiguration()?.update(
-        VERBOSE_LOGS,
-        true,
-        ConfigurationTarget.Global,
-    )
-    window.showInformationMessage("Verbose logging enabled.")
+export function isNotificationEnabled(): boolean {
+    return getSonarLintConfiguration().get(NOTIFY_COMPILE_COMMANDS, true)
 }
 
-export function loadInitialSettings() {
-    currentConfig = getSonarLintConfiguration()
+export function updateNotificationDisabled(value: boolean, target?: ConfigurationTarget | undefined) {
+    getSonarLintConfiguration().update(NOTIFY_COMPILE_COMMANDS, value, target)
 }
 
-export function getCurrentConfiguration() {
-    return currentConfig
+export function updateCompileCommandsPath(value: string | undefined, target?: ConfigurationTarget | undefined) {
+    getSonarLintConfiguration().update(PATH_TO_COMPILE_COMMANDS, value, target)
+}
+
+export function updateVerboseLogging(value: boolean, target?: ConfigurationTarget | undefined) {
+    getSonarLintConfiguration().update(VERBOSE_LOGS, value, target)
 }
 
 export function onConfigurationChange() {

